store/tikv/prewrite.go

```diff
@@ -162,6 +168,7 @@ func (action actionPrewrite) handleSingleBatch(c *twoPhaseCommitter, bo *Backoff
 						zap.Uint64("startTS", c.startTS))
 					tikvOnePCTxnCounterFallback.Inc()
 					c.setOnePC(false)
+					c.setAsyncCommit(false)
```

Comment:
Why it's here, instead of in the `if` clause at L190?

Reply:
When using 1pc, `useAsyncCommit` may be true. This field should be set false when fallback from 1pc. 
Otherwise, we will get into the wrong branch in https://github.com/pingcap/tidb/blob/0eb8ff97bc/store/tikv/2pc.go#L1162 and https://github.com/pingcap/tidb/blob/0eb8ff97bc/store/tikv/2pc.go#L1235

Reply:
Oh sorry. I just notice that L196 has already unset async commit for non-1pc case.  But for 1pc case I recommend that also check `min_commit_ts` field to be more strict. There's current no case that 1pc may fallback to async commit though.